### PR TITLE
758 provide more useful messages in the UI when a session is being created

### LINKED ISSUE #708: Add error log to capture logs when an error occur during the ssh process
### Is your feature request related to a problem? Please describe.

Currently when an error occurs user only sees failed to authenticate.

### Describe the solution you'd like

Update the SSHNP Class and Terminal Session Controller for capture error logs.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #709: Add configuration option for --no-encrypt-rvd-traffic
### Is your feature request related to a problem? Please describe.

 user cannot configure the --no-encrypt-rvd-traffic parameter

### Describe the solution you'd like

 when creating a profile the user should be able to configure the --no-encrypt-rvd-traffic parameter.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #758: Provide more useful messages in the UI when a session is being created.
### Is your feature request related to a problem? Please describe.

Currently, generic messages like "Starting Shell Session..." is displayed in the terminal. User should get more meaningful messages.

### Describe the solution you'd like

User should get more meaningful messages.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CurtlyCritchlow**
**- What I did**
Fixes #758 
Fixes #709 
**- How I did it**
-  Used progressStream to provide user with useful messages in the UI
- Added authenticateClientToRvd, authenticateDeviceToRVD and encryptedRvdTraffic to the profile form
- Updated tooltip message use same message as the cli for Sshnp arguments

**- How to verify it**

1. Run the app 
2. navigate to profile form screen, the new form screen should be present
3. try to ssh, the terminal screen should display more useful information to the user.

**- Description for the changelog**

- Updated profile form
- updated messages shows to user during ssh

### CODE REVIEW

**Comment by xavierchanth** [comment]
@CurtlyCritchlow 

While it makes sense to say "when false, \<negative effects\>" for the CLI because they are on by default. In the UI we should always use the positive: "when turned on, \<positive effects\>" since we have a visual indication showing that these are on.

File: packages/dart/sshnp_flutter/pubspec.yaml [hunk 1]
```diff
@@ -31,12 +31,12 @@ dependencies:
   intl: any
   macos_ui: ^2.0.0
   mocktail: ^1.0.0
-  noports_core: ^5.0.0
+  noports_core: ^6.0.0
```
Comment (xavierchanth) [inline packages/dart/sshnp_flutter/pubspec.yaml]: noports_core: ^6.0.1

**Review by xavierchanth — DISMISSED** [review]

**Comment by xavierchanth** [comment]
Otherwise LGTM

**Review by gkc — APPROVED** [review]

**Review by gkc — CHANGES_REQUESTED** [review]
Still need to update noports_core to 6.0.1

File: packages/dart/sshnp_flutter/pubspec.yaml (hunk 1 above)
Comment (gkc) [reply packages/dart/sshnp_flutter/pubspec.yaml]: @CurtlyCritchlow still need to change the noports_core dependency

**Comment by CurtlyCritchlow** [comment]
> Still need to update noports_core to 6.0.1

because I'm using "^" before 6.0.0 it will use the latest which is 6.0.1. Do you still want to explicitly use 6.0.1?

**Comment by xavierchanth** [comment]
> > Still need to update noports_core to 6.0.1
> 
> because I'm using "^" before 6.0.0 it will use the latest which is 6.0.1. Do you still want to explicitly user 6.0.1?

In this case, best to pin to 6.0.1

**Review by xavierchanth — APPROVED** [review]

**Comment by CurtlyCritchlow** [comment]
@gkc plese review to unblock merge.

**Review by gkc — APPROVED** [review]

**Pull request merged**